src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -360,6 +358,53 @@ private static void AddProjectFileDependenciesForPackageReference(PackageSpec pr
             }
         }
 
+        private void AddProjectCentralTransitiveDependencyGroupsForPackageReference(PackageSpec project, LockFile lockFile, IEnumerable<RestoreTargetGraph> targetGraphs)
+        {
+            // TargetFrameworkInformations should have only distinct FrameworkNames
+            var centralPackageVersionsPerFramework = project.TargetFrameworks.ToDictionary(tfmi => tfmi.FrameworkName, tfmi => tfmi.CentralPackageVersions);
+
+            foreach (var targetGraph in targetGraphs.OrderBy(graph => graph.Framework.ToString(), StringComparer.Ordinal))
+            {
+                if (_includeFlagGraphs.TryGetValue(targetGraph, out Dictionary<string, LibraryIncludeFlags> dependenciesIncludeFlags))
+                {
+                    // The transitive dependencies enforced by the central package version management file are written to the assets to be used by the pack task.
+                    IEnumerable<LibraryDependency> centralEnforcedTransitiveDependencies = targetGraph
+                        .Flattened
+                        .Where(graphItem => graphItem.IsCentralTransitive
+                            && centralPackageVersionsPerFramework.ContainsKey(targetGraph.Framework)
+                            && centralPackageVersionsPerFramework[targetGraph.Framework].ContainsKey(graphItem.Key.Name))
+                        .Select((graphItem) =>
+                        {
+                            var matchingCentralVersion = centralPackageVersionsPerFramework[targetGraph.Framework][graphItem.Key.Name];
+                            var libraryDependency = new LibraryDependency()
+                            {
+                                LibraryRange = new LibraryRange(matchingCentralVersion.Name, matchingCentralVersion.VersionRange, LibraryDependencyTarget.Package),
+                                ReferenceType = LibraryDependencyReferenceType.Transitve,
+                                VersionCentrallyManaged = true
+                            };
+
+                            if (dependenciesIncludeFlags.TryGetValue(matchingCentralVersion.Name, out LibraryIncludeFlags libraryIncludeFlags))
```

Comment:
What do we do if it's not there. It's impossible, correct?
I'd just do a get instead of a try get.

Reply:
Correct - the tryget was more defensive. Updated to get the value and let it throw.

Reply:
Given that this is a completely unexpected thing from which we can't really recover graciously, we should just fail fast.

That'd be a lot easier to diagnose than an unexpected value in the flags :)

Reply:
Updated.